ste/JobPartPlan.go

```diff
@@ -144,11 +145,20 @@ func (jpph *JobPartPlanHeader) TransferSrcDstStrings(transferIndex uint32) (sour
 	sh.Cap = sh.Len
 	dstRelative := string(dstSlice)
 
+	// Incase user is trying to copy complete directory in windows i.e D:\\
```

Comment:
It feels a bit odd to do this processing here, inside the JobPartPlanHeader. I believe it would be more consistent, with the way paths are usually handled in AzCopy, to do this processing at the point where the root directory is first set up. Which is here: https://github.com/Azure/azure-storage-azcopy/blob/9a88118d55716be9408a2d33fd1a0cffd50f9221/cmd/copyEnumeratorInit.go#L535

What do you think about moving the processing to that point?